src/main/java/org/springframework/data/couchbase/config/AbstractCouchbaseConfiguration.java

```diff
@@ -130,6 +140,10 @@ public Cluster couchbaseCluster(ClusterEnvironment couchbaseClusterEnvironment)
 	@Bean(destroyMethod = "shutdown")
 	public ClusterEnvironment couchbaseClusterEnvironment() {
 		ClusterEnvironment.Builder builder = ClusterEnvironment.builder();
+		if (!nonShadowedJacksonPresent()) {
+			throw new CouchbaseException("non-shadowed Jackson not present");
```

Comment:
is there a reason we need to blow up if it is not present? (or just add the custom serializer if present?) -  could log a warning instead if unlikely?

Reply:
Good question. In the SDK, when the non-shadowed jackson classes are not present, a DefaultJsonSerializer is created.  DefaultJsonSerializer has no means to use an already-created mapper, or expose the mapper that it creates.  I suppose this would only be an issue when the user wants to us couchbaseObjectMapper() and expects it to be the same one used by the serializer.  However, David said the not having the non-shadowed classes in Spring is impossible.  The alternative would be to issue a warning saying that the serializer was not using couchbaseObjectMapper() and continue.

   return nonShadowedJacksonPresent()
        ? JacksonJsonSerializer.create(cryptoManager)
        : DefaultJsonSerializer.create(cryptoManager);